### PR TITLE
Social login

### LINKED ISSUE #7: Social login
As a user I should be able to login with social login such as Github, Google.

NextAuth.js support social login.

- Add social login button to the login and signup page.
- Update the README with the appropriate instructions.

### PR DESCRIPTION
**Pull request by Saptarshee94**
This Pull request adds social login buttons for **GitHub** and **Google** in `Login` and `Join` page.
Resolves #7 
`README.md` modified.
_Prerequisite: Need to set up **Google** and **GitHub** OAuth applications and use generated credentials in `.env` file._

### CODE REVIEW

File: pages/auth/join.tsx [hunk 1]
```diff
@@ -31,6 +35,31 @@ const Signup: NextPageWithLayout<inferSSRProps<typeof getServerSideProps>> = ({
           <Join />
         )}
       </div>
+      <div className="divider">or sign up with</div>
+      <Link href="/api/auth/signin">
```
Comment (devkiran) [inline pages/auth/join.tsx:39]: Do we need both Link href and onClick action?

File: .env.example [hunk 2]
```diff
@@ -14,4 +14,10 @@ DATABASE_URL=
 APP_URL=http://localhost:4002
 
 SVIX_URL=https://api.eu.svix.com
-SVIX_API_KEY=
\ No newline at end of file
+SVIX_API_KEY=
+
+GITHUB_ID=
```
Comment (devkiran) [inline .env.example:19]: Rename this to GITHUB_CLIENT_ID and Rename this to GITHUB_CLIENT_SECRET

File: pages/auth/login.tsx [hunk 3]
```diff
@@ -104,6 +106,31 @@ const Login: NextPageWithLayout<
...
+          <Link href="/api/auth/signin">
+            <a
+              className="btn btn-outline w-full"
+              onClick={(e) => {
+                e.preventDefault();
+                signIn("github");
+              }}
+            >
+              <FaGithub />
+              &nbsp;Sign in with Github
```
Comment (devkiran) [inline pages/auth/login.tsx:131]: The text should be **GitHub**

Comment (Saptarshee94) [reply pages/auth/login.tsx]: Resolved typos.

File: .env.example (hunk 2 above)
Comment (Saptarshee94) [reply .env.example]: Resolved.

File: pages/auth/join.tsx (hunk 1 above)
Comment (Saptarshee94) [reply pages/auth/join.tsx:39]: `onClick` action is required for the operation, `href` prop in mandatory for the `Link` component.
A NextAuth.js example using both `href` and `onclick` event: [Example](https://github.com/nextauthjs/next-auth-example/blob/main/components/header.tsx)

Comment (devkiran) [reply pages/auth/join.tsx:39]: Can we use button instead?

Comment (Saptarshee94) [reply pages/auth/join.tsx]: Resolved, updated links to buttons.

**Comment by devkiran** [comment]
I tested this locally with GitHub Auth and got the `OAuthAccountNotLinked` error after the redirect from GitHub. So it looks like this requires more work on the NextAuth side. Any thoughts?

**Comment by Saptarshee94** [comment]
> I tested this locally with GitHub Auth and got the `OAuthAccountNotLinked` error after the redirect from GitHub. So it looks like this requires more work on the NextAuth side. Any thoughts?

I am not sure, but according to next auth `OAuthAccountNotLinked` error occurs due to following reason.

> OAuthAccountNotLinked: If the email on the account is already linked, but not with this OAuth account.

I think if the apps auth requires more work to tackle the said issue, It can be addressed as a new issue. If you found any solution or feeback just let me know.

**Comment by devkiran** [comment]
I tried with a new database, and I see a new user is created but got the same `OAuthAccountNotLinked` error. So the login was not successful.

**Comment by devkiran** [comment]
I tried with a new database, and the login was not successful. I got the following error.

Error

**Comment by devkiran** [comment]
Please test it from your side and see if you can reproduce this issue. I believe you have to adjust the code in `[...nextauth].ts` to make this work.

**Comment by Saptarshee94** [comment]
> Please test it from your side and see if you can reproduce this issue. I believe you have to adjust the code in `[...nextauth].ts` to make this work.

Able to reproduce the issue but facing several issues while making changes in [...nextauth.ts] such as `OauthNotLinked` error `OauthCreateAccount` error. Need your help to point out what to change.

**Review by devkiran — APPROVED** [review]

**Pull request merged**